src/connectors/paymentService.ts

```diff
@@ -1072,4 +1078,102 @@ export class PaymentService extends BaseService {
       })
       .returning('*')
   }
+
+  /*********************************
+   *                               *
+   *           notification        *
+   *                               *
+   *********************************/
+
+  notifyDonation = async ({
+    tx,
+    sender,
+    recipient,
+    article,
+  }: {
+    tx: Transaction
+    sender: User
+    recipient: User
+    article: {
+      title: string
+      slug: string
+      authorId: string
+      mediaHash: string
+      draftId: string
+    }
+  }) => {
+    const atomService = new AtomService()
+    const notificationService = new NotificationService()
+    const amount = parseFloat(tx.amount)
+    // send email to sender
+    const author = await atomService.findFirst({
+      table: 'user',
+      where: { id: article.authorId },
+    })
+    const draft = await atomService.findFirst({
+      table: 'draft',
+      where: { id: article.draftId },
+    })
+
+    const hasReplyToDonator =
+      draft.replyToDonator !== undefined &&
+      draft.replyToDonator !== null &&
+      draft.replyToDonator !== ''
+    const _article = {
+      id: tx.targetId,
+      title: article.title,
+      slug: article.slug,
+      mediaHash: article.mediaHash,
+      author: {
+        displayName: author.displayName,
+        userName: author.userName,
+      },
+      hasReplyToDonator,
+    }
+
+    notificationService.mail.sendPayment({
+      to: sender.email,
+      recipient: {
+        displayName: sender.displayName,
+        userName: sender.userName,
+      },
+      type: 'donated',
+      article: _article,
```

Comment:
can simply like below, then `_article` is unnecessary.

```tsx
article: { ...article, id: tx.targetId, hasReplyToDonator }
```

Reply:
not all article fields （especially content field） need to pass to sendGrid api?